bzl/toolchains/BUILD

```diff
@@ -0,0 +1,247 @@
+# Heavily inspired by pico-sdk's toolchain configuration,
+# although it didn't particularly work.
+load("@pico-sdk//bazel/toolchain:configurable_feature.bzl", "configurable_toolchain_feature")
+load("@rules_cc//cc/toolchains:args.bzl", "cc_args")
+load("@rules_cc//cc/toolchains:args_list.bzl", "cc_args_list")
+load("@rules_cc//cc/toolchains:feature.bzl", "cc_feature")
+load("@rules_cc//cc/toolchains:toolchain.bzl", "cc_toolchain")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_args(
+    name = "armv6m-none-eabi",
+    actions = [
+        "@rules_cc//cc/toolchains/actions:compile_actions",
+        "@rules_cc//cc/toolchains/actions:link_actions",
+    ],
+    args = ["--target=armv6m-none-eabi"],
+)
+
+cc_args(
+    name = "xtensa",
+    actions = [
+        "@rules_cc//cc/toolchains/actions:compile_actions",
+        "@rules_cc//cc/toolchains/actions:link_actions",
+    ],
+    args = [
+        # TODO: Figure out what flags are needed from esp-idf.
+        # "-mcpu=cortex-m0plus",
+        # "-mthumb",
+    ],
+)
+
+cc_args(
+    name = "no-canonical-system-headers",
+    actions = ["@rules_cc//cc/toolchains/actions:compile_actions"],
+    args = ["-fno-canonical-system-headers"],
+)
+
+cc_args(
+    name = "no-canonical-prefixes",
+    actions = ["@rules_cc//cc/toolchains/actions:compile_actions"],
+    args = ["-no-canonical-prefixes"],
+)
+
+cc_args(
+    name = "nostdlibxx",
+    actions = ["@rules_cc//cc/toolchains/actions:link_actions"],
+    args = [
+        "-nostdlib++",
+    ],
+)
+
+cc_args(
+    name = "nostartfiles",
+    actions = ["@rules_cc//cc/toolchains/actions:link_actions"],
+    args = ["-nostartfiles"],
+)
+
+# Hack to get the system includes to work; these paths are relative
+# to the repo root.
+cc_args(
+    name = "sys_includes",
+    actions = ["@rules_cc//cc/toolchains/actions:compile_actions"],
+    args = [
+        "-Itoolchains/esp_clang_linux-x86_64/xtensa-esp32s3-elf/include/",
+        "-Itoolchains/esp_clang_linux-x86_64/xtensa-esp32s3-elf/include/c++/11.2.0/",
+        "-Itoolchains/esp_clang_linux-x86_64/xtensa-esp32s3-elf/include/c++/11.2.0/backward",
+        "-Itoolchains/esp_clang_linux-x86_64/xtensa-esp32s3-elf/include/c++/11.2.0/xtensa-esp32s3-elf",
+    ],
+)
```

Comment:
It was terribly annoying to find a hermetic workaround, but this does do the trick, provided that every `cc_library` / `cc_binary` depends on `//toolchains:esp_clang_sys_includes`